src/amplitude-client.js

```diff
@@ -894,7 +896,9 @@ AmplitudeClient.prototype._logEvent = function _logEvent(eventType, eventPropert
     _saveCookieData(this);
 
     userProperties = userProperties || {};
-    apiProperties = apiProperties || {};
+    var trackingOptions = _generateApiPropertiesTrackingConfig(this);
```

Comment:
To keep event payload as small as possible, only send the location fields that have been disabled. In theory we can generate this once and save it to the instance, although if the user manually overrides `amplitude.getInstance().options.trackingOptions.<field>` after `init`, then regenerating this every event would capture that? Although seems kind of excessive @blazzy  what you think?

Reply:
Mutating the options object off the instance isn't a supported way of interacting with the SDK, I think we can pre-generate this.